docs/rfcs/component-universal-telemetry.md

```diff
@@ -0,0 +1,212 @@
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.exporter.consumed.items:
+      enabled: true
+      description: Number of items passed to the exporter.
+      unit: "{items}"
+      sum:
+        value_type: int
+        monotonic: true
+
+   otelcol.receiver.produced.size:
+      enabled: false
+      description: Size of items emitted from the receiver.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.processor.consumed.size:
+      enabled: false
+      description: Size of items passed to the processor.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.processor.produced.size:
+      enabled: false
+      description: Size of items emitted from the processor.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.connector.consumed.size:
+      enabled: false
+      description: Size of items passed to the connector.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.connector.produced.size:
+      enabled: false
+      description: Size of items emitted from the connector.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+    otelcol.exporter.consumed.size:
+      enabled: false
+      description: Size of items passed to the exporter.
+      unit: "By"
+      sum:
+        value_type: int
+        monotonic: true
+```
+
+### Auto-Instrumented Logs
+
+Metrics provide most of the observability we need but there are some gaps which logs can fill. Although metrics would describe the overall
+item counts, it is helpful in some cases to record more granular events. e.g. If a produced batch of 10,000 spans results in an error, but
+100 batches of 100 spans succeed, this may be a matter of batch size that can be detected by analyzing logs, while the corresponding metric
+reports only that a 50% success rate is observed.
+
+For security and performance reasons, it would not be appropriate to log the contents of telemetry.
+
+It's very easy for logs to become too noisy. Even if errors are occurring frequently in the data pipeline, they may only be of interest to
+many users if they are not handled automatically.
+
+With the above considerations, this proposal includes only that we add a DEBUG log for each individual outcome. This should be sufficient for
+detailed troubleshooting but does not impact users otherwise.
+
+In the future, it may be helpful to define triggers for reporting repeated failures at a higher severity level. e.g. N number of failures in
+a row, or a moving average success %. For now, the criteria and necessary configurability is unclear so this is mentioned only as an example
+of future possibilities.
+
+### Auto-Instrumented Spans
+
+It is not clear that any spans can be captured automatically with the proposed mechanism. We have the ability to insert instrumentation both
```

Comment:
> Do we accept that the span is just measuring a quick handoff to the internal state of the component? 

Yes: two incoming spans from different connections would belong to two different traces. They would be assembled into one batch and exporter as part of a single outgoing call by the exporter. This to me means that we have three traces: two representing the original requests, and one starting with the batch processor and ending with the exporter call. The entire flow can be understood by looking at the span links.

> Is this going to be misleading to users?

Good question: the scenario I described above is what I would expect as a user, but not sure everybody sees it the same way.

Do you think we can refine this later, or is it important to answer as part of this PR?

Reply:
I generally agree with what you're saying but I agree it would be best handled later, and perhaps authored by someone other than myself who has a more complete understanding of tracing best practices. I'll leave this in place for now but create a tracking issue to reconsider this section.

Reply:
I've opened https://github.com/open-telemetry/opentelemetry-collector/issues/11743 to track this